docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
+
+* be used to hold the git submodule reference to the upstream repository.
+* contain a Containerfile that references and pulls in the git submodule(s) content.
+* become a Component of our {ProductName} application.
+
+=== Create the git submodule for the upstream repo to track it
+
+An **upstream repo** is a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+  branch = main # Ensure the branch is correct
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+=== Onboard the project onto {ProductName}
+
+After creating the repository, onboard to {ProductName} and configure your application and its components.
```

Comment:
It may not be a new application. I think we can probably just say to onboard the component and xref to the relevant docs.

Reply:
Ack, I referenced the "Creating applications and components" doc